File: src/main/java/org/zenbot/szolnok/timetable/dao/BusRepository.java
package org.zenbot.szolnok.timetable.dao;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenbot.szolnok.timetable.domain.Bus;

public interface BusRepository extends MongoRepository<Bus, String> {

    Bus findByBusName(String busName);
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/dao/BusRepository.kt
package org.zenbot.szolnok.timetable.dao

import org.springframework.data.mongodb.repository.MongoRepository
import org.zenbot.szolnok.timetable.domain.Bus
import java.util.Optional

interface BusRepository : MongoRepository<Bus, String> {

    fun findByBusName(busName: String): Optional<Bus>
}
